include/wlr/types/wlr_xdg_shell.h

```diff
@@ -121,10 +125,11 @@ struct wlr_xdg_toplevel {
 	struct wlr_xdg_surface *parent;
 	struct wl_listener parent_unmap;
 
-	struct wlr_xdg_toplevel_state client_pending;
-	struct wlr_xdg_toplevel_state server_pending;
-	struct wlr_xdg_toplevel_state last_acked;
-	struct wlr_xdg_toplevel_state current;
+	struct wlr_xdg_toplevel_state current, pending;
+	struct wlr_xdg_toplevel_state scheduled;
```

Comment:
I think some more verbose docs here would go a long way towards making this code easier to understand for wlroots contributors and compositor authors alike. Feel free to expand/tweak as you see fit, or just copy this verbatim:
```suggestion
	// The current and pending state of the toplevel. The pending state
	// includes values sent by the compositor in a configure as well as
	// the max/min width/height set by the client. When the client commits
	// the surface, the entire pending state is copied as-is into the
	// current state.
	struct wlr_xdg_toplevel_state current, pending;
	// State scheduled by the compositor to be sent to the client in
	// the next configure event. The max/min width/height fields are ignored
	struct wlr_xdg_toplevel_state scheduled;
```

Reply:
Added a comment for `scheduled` state. I don't think there is a reason to describe `current`/`pending`, as it basically copies `wlr_surface.{current,pending}` semantics.